src/apps/velodrome/optimism/velodrome.bribe.contract-position-fetcher.ts

```diff
@@ -0,0 +1,101 @@
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import { ContractPositionTemplatePositionFetcher } from '~position/template/contract-position.template.position-fetcher';
+import {
+    DefaultContractPositionDefinition,
+    GetDisplayPropsParams,
+    GetTokenBalancesParams,
+} from '~position/template/contract-position.template.types';
+
+import { VelodromeContractFactory, VelodromeBribe } from '../contracts';
+
+interface VelodromeApiPairData {
+    gauge: GaugeData;
+}
+
+interface GaugeData {
+    wrapped_bribe_address: string;
+}
+
+@PositionTemplate()
+export class OptimismVelodromeBribeContractPositionFetcher extends ContractPositionTemplatePositionFetcher<VelodromeBribe> {
+    constructor(
+        @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+        @Inject(VelodromeContractFactory) protected readonly contractFactory: VelodromeContractFactory,
+    ) {
+        super(appToolkit);
+    }
+    veTokenAddress = '0x9c7305eb78a432ced5c4d14cac27e8ed569a2e26';
+    veVoteAddress = '0x09236cff45047dbee6b921e00704bed6d6b8cf7e';
+    groupLabel = 'Bribe';
+
+    getContract(address: string): VelodromeBribe {
+        return this.contractFactory.velodromeBribe({ address, network: this.network });
+    }
+
+    async getDefinitions(): Promise<DefaultContractPositionDefinition[]> {
+        const { data } = await axios.get<{ data: VelodromeApiPairData[] }>('https://api.velodrome.finance/api/v1/pairs');
+        var definitions = new Array;
+        data.data.filter(v => !!v).filter(v => !!v.gauge).forEach(pool => {
+            const wBribeAddress = pool.gauge.wrapped_bribe_address;
+            if (wBribeAddress != null) {
+                definitions.push({ address: wBribeAddress });
+            }
+        }
+        );
+        return definitions;
+    }
+
+    async getTokenDefinitions({ contract }) {
+        const numRewards = Number(await contract.rewardsListLength());
+        const bribeTokens = await Promise.all(range(numRewards).map(async n => await contract.rewards(n)));
+        const baseTokens = await this.appToolkit.getBaseTokenPrices(this.network);
+        var tokenDefinitions = new Array();
+        for (let token of bribeTokens) {
+            const tokenFound = baseTokens.find(p => p.address === token);
+            if (tokenFound) {
+                tokenDefinitions.push({
+                    metaType: MetaType.CLAIMABLE,
+                    address: token,
+                    network: this.network,
+                });
+            }
+        }
+        return tokenDefinitions;
+    }
+
+    async getLabel({ contractPosition }: GetDisplayPropsParams<VelodromeBribe>): Promise<string> {
+        return `${contractPosition.address} Bribe`;
+    }
+
+    async getTokenBalancesPerPosition({ address, contract, contractPosition }: GetTokenBalancesParams<VelodromeBribe>): Promise<BigNumberish[]> {
+        const multicall = this.appToolkit.getMulticall(this.network);
```

Comment:
You don't need to redeclare multicall, you can use the one already declared in `GetTokenBalancesParams`

async getTokenBalancesPerPosition({ address, contract, contractPosition, multicall }:

Reply:
Sure